pkg/operator/controllers/pullsecret/pullsecret_controller_test.go

```diff
@@ -285,11 +286,11 @@ func TestPullSecretReconciler(t *testing.T) {
 			ctx := context.Background()
 
 			clientFake := ctrlfake.NewClientBuilder().WithObjects(tt.instance).WithObjects(tt.secrets...).Build()
```

Comment:
We should also add validation in these tests to ensure that the conditions on the controller are what we expect (e.g. Degraded:True when an error occurs). You can see an example of this in the dnsmasq machine config controller tests: https://github.com/Azure/ARO-RP/blob/master/pkg/operator/controllers/dnsmasq/machineconfig_controller_test.go#L36

Reply:
This comment has been addressed

Reply:
I think some work is still needed here. Even though we now validate that the conditions are what we expect, there are no test cases covering the cases where the controller becomes degraded.

So, for example, we could add a test case where the reconciler fails to update the ARO Cluster resource toward the end of the `Reconcile` function, and this test case could then verify that the degraded condition is set properly.

I looked into ways to get that `Update` function to return an error, and I happened to come across an article written by a Red Hatter that looks like it'll help: https://www.bater.io/posts/2020/06/throwing-errors-with-controller-runtime-fake/